src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -403,6 +411,31 @@ public static Vector256<ushort> Load256(ref byte ptr)
                 (Vector128<ushort> lower, Vector128<ushort> upper) = Vector128.Widen(Vector128.LoadUnsafe(ref ptr));
                 return Vector256.Create(lower, upper);
             }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            [CompExactlyDependsOn(typeof(Avx))]
+            public static bool Compare256(ref byte utf8, ref ushort utf16)
+            {
+                // We widen the utf8 param so we can compare it to utf16, this doubles how much of the utf16 vector we search
+                Debug.Assert(Vector256<byte>.Count == Vector256<ushort>.Count * 2);
+
+                Vector256<byte> leftNotWidened = Vector256.LoadUnsafe(ref utf8);
+                if (!AllCharsInVectorAreAscii(leftNotWidened))
+                {
+                    return false;
+                }
+
+                (Vector256<ushort> lower, Vector256<ushort> upper) = Vector256.Widen(leftNotWidened);
+                Vector256<ushort> rightValues0 = Vector256.LoadUnsafe(ref utf16);
+                Vector256<ushort> rightValues1 = Vector256.LoadUnsafe(ref utf16, (uint)Vector256<ushort>.Count);
+
+                if (((lower ^ rightValues0) | (upper ^ rightValues1)) != Vector256<ushort>.Zero)
```

Comment:
very subjective nit: "lower" and "upper" names don't suggest that these belong to "left", moreover using number-based suffixes always sickens my inner "clean coder"

```suggestion
                (Vector256<ushort> leftLower, Vector256<ushort> leftUpper) = Vector256.Widen(leftNotWidened);
                Vector256<ushort> right = Vector256.LoadUnsafe(ref utf16);
                Vector256<ushort> rightNext = Vector256.LoadUnsafe(ref utf16, (uint)Vector256<ushort>.Count);

                if (((leftLower ^ right) | (leftUpper ^ rightNext)) != Vector256<ushort>.Zero)
```